### ISSUE TITLE
ECS Autoscaling Cluster does not recognize containerOverrides

### ISSUE THREAD

**Issue by jack-michaud** [aws/aws-cli#4739]
Not sure if this is the right place for this as I first found this in the JS SDK, but I can repro on the CLI. I think this could be a serverside issue.
When launching a new task with container overrides onto an ECS autoscaling cluster with no instances in it (created from [this guide](https://aws.amazon.com/blogs/aws/aws-ecs-cluster-auto-scaling-is-now-generally-available/)) they are not recognized.

## Reproduction
Here's a repro... `poc` is just an task definition that echos out an environment variable. The task definition is actually irrelevant because it seems like the overrides aren't registered by AWS.

Here's my command, run when the cluster is at 0 instances:
```
$ aws ecs run-task --cluster my-autoscaling-cluster --task-definition poc:2 --overrides file://overrides.json
{
    "tasks": [
        {
            "attachments": [],
            "capacityProviderName": "my-autoscaling-cluster-capacity-provider",
            "clusterArn": "arn:aws:ecs:us-west-1:*****:cluster/my-autoscaling-cluster",
            "containers": [],
            "cpu": "128",
            "createdAt": 1575852294.031,
            "desiredStatus": "RUNNING",
            "group": "family:poc",
            "lastStatus": "PROVISIONING",
            "launchType": "EC2",
            "memory": "128",
            "overrides": {
                "containerOverrides": [],
                "inferenceAcceleratorOverrides": []
            },
            "tags": [],
            "taskArn": "arn:aws:ecs:us-west-1:*****:task/my-autoscaling-cluster/400f999dd8114f12ad8378931dd69d98",
            "taskDefinitionArn": "arn:aws:ecs:us-west-1:*****:task-definition/poc:2",
            "version": 1
        }
    ],
    "failures": []
}
```

`overrides.json`
```
{
  "containerOverrides": [
    {
      "name": "poc",
      "environment": [
        {
          "name": "MYVAR",
          "value": "Please read me!"
        }
      ]
    }
  ]
}
```

Note the empty `containerOverrides` key from the `run-task` output.

After the EC2 instance is provisioned and the task is run, the resultant task indeed does not register the environment variables I specified.

![image](https://user-images.githubusercontent.com/13189762/70400870-e0ea2a00-19e1-11ea-9227-aca30f04c1e7.png)

Here's me running the same `run-task` command after instances have been provisioned:
```
$ aws ecs run-task --cluster my-autoscaling-cluster --task-definition poc:2 --overrides file://overrides.json
{ 
    "tasks": [ 
        { 
            "attachments": [], 
            "availabilityZone": "us-west-1b", 
            "capacityProviderName": "my-autoscaling-cluster-capacity-provider-1", 
            "clusterArn": "arn:aws:ecs:us-west-1:*****:cluster/my-autoscaling-cluster", 
            "containerInstanceArn": "arn:aws:ecs:us-west-1:*****:container-instance/my-autoscaling-cluster/c10655569f6240e6b6e6af11faa823d6", 
            "containers": [ 
                { 
                    "containerArn": "arn:aws:ecs:us-west-1:*****:container/e5d7a383-343c-4e7c-a3d0-557cb16abd44", 
                    "taskArn": "arn:aws:ecs:us-west-1:*****:task/my-autoscaling-cluster/bc4e25ff300e4f2991ef1f426b837af3", 
                    "name": "poc", 
                    "image": "alpine", 
                    "lastStatus": "PENDING", 
                    "networkInterfaces": [], 
                    "cpu": "128", 
                    "memory": "128" 
                } 
            ], 
            "cpu": "128", 
            "createdAt": 1575852605.568, 
            "desiredStatus": "RUNNING", 
            "group": "family:poc", 
            "lastStatus": "PENDING", 
            "launchType": "EC2", 
            "memory": "128", 
            "overrides": { 
                "containerOverrides": [ 
                    { 
                        "name": "poc", 
                        "environment": [ 
                            { 
                                "name": "MYVAR", 
                                "value": "Please read me!" 
                            } 
                        ] 
                    } 
                ], 
                "inferenceAcceleratorOverrides": [] 
            }, 
            "tags": [], 
            "taskArn": "arn:aws:ecs:us-west-1:*****:task/my-autoscaling-cluster/bc4e25ff300e4f2991ef1f426b837af3", 
            "taskDefinitionArn": "arn:aws:ecs:us-west-1:*****:task-definition/poc:2", 
            "version": 1 
        } 
    ], 
    "failures": [] 
} 
```

`containerOverrides` looks successfully percolated.

![image](https://user-images.githubusercontent.com/13189762/70399875-5fdc6400-19dc-11ea-8433-a3f75e4fd59a.png)

**Comment by jamesls** [comment]
From what it looks like, that appears to be an issue on the service side, especially given you're seeing this in both the AWS CLI and the JS SDK.  I can forward this to the service team, but I would also suggest asking on the ECS forums (https://forums.aws.amazon.com/forum.jspa?forumID=187).

I'm going to go ahead and close out this issue as we aren't able to fix this in the CLI.  Let me know if you have any other questions.

**Comment by NielsCorneille** [comment]
Posted a bug report on the ECS forums: https://forums.aws.amazon.com/thread.jspa?threadID=314493

**Comment by jack-michaud** [comment]
Thanks @NielsCorneille. 
FWIW I also created a ticket with AWS through my company's support plan. Last response was on 12/11 saying the ECS team is "actively investigating this issue".

**Comment by jack-michaud** [comment]
Oh, in the meantime, I came up with a temporary workaround: I have a small wrapper script that pulls the arguments that would normally be provided through environment variables from an SQS queue. 
One catch: be careful about cancelling a `PROVISIONING` task that hasn't pulled its message off the queue because it will orphan that message. The next task will run that message and the intended task will remain on the queue. I just have application logic that prevents deleting the task, but it's really not ideal. 
Hope that unblocks some people.

**Comment by robberry** [comment]
Thanks for posting this - I have the same issue it seems.  I will lodge a support request as well.

**Comment by solutionrooms** [comment]
I have the same issue exactly either starting at 0 or any number of instances using auto scaling. the boto3 run_task command override is ignored for any tasks that are waiting for provisioning. However after the instances are provisioned and registered to ecs the same boto3 command works (the override is accepted)

Is there any update or ETA for this important ECS bug?
